client/webpack.config.js

```diff
@@ -246,6 +246,7 @@ const webpackConfig = {
 	resolve: {
 		extensions: [ '.json', '.js', '.jsx', '.ts', '.tsx' ],
 		mainFields: [ 'browser', 'calypso:src', 'module', 'main' ],
+		conditionNames: [ 'calypso:src', 'import', 'module', 'require' ],
```

Comment:
Docs: https://webpack.js.org/configuration/resolve/#resolveconditionnames

This tells webpack which values to look for in conditional exports. Note the order here doesn't matter (according to my experiments, I didn't find any relevant docs):

* `calypso:src` is the new conditional export I introduced, equivalent to the old `pkg['calypso:src']`.
* The rest is a concatenation of https://github.com/webpack/webpack/blob/main/lib/config/defaults.js